workers/data_refinery_workers/processors/transcriptome_index.py

```diff
@@ -155,6 +155,30 @@ def _extract_assembly_information(job_context: Dict) -> Dict:
             assembly_name_start_index = versionless_url.rfind(".") + 1
             job_context["assembly_name"] = versionless_url[assembly_name_start_index:]
 
+            database_name = "Ensembl"
+
+            # The division name follows right after the first occurence of the assembly version (is there a better way to do this?)
+            try:
+                division_name = versionless_url.split(job_context["assembly_version"])[1][1:].split(
+                    "/"
+                )[0]
+
+                # If the url is for the main Ensembl then it won't be any of the following
+                if division_name not in ["plants", "metazoa", "fungi", "bacteria", "protists"]:
+                    database_name = "EnsemblMain"
+                else:
+                    database_name = "Ensembl" + str.capitalize(division_name)
+            except:
+                # This may be a test without a source_url
+                if not og_file.source_url:
+                    logger.info(
+                        "Failed to retrieve/check for division name from url (looks like there was no source_url for the original file)"
+                    )
+                else:
+                    logger.error("Failed to retrieve/check for division name from url")
```

Comment:
Hmm, I think it might better to fail the job, so we know that we need to fix something. It also sounds like the tests should set `source_url` on the original file so they don't have to be handled specially.